tests/service_postgres.py

```diff
@@ -0,0 +1,307 @@
+    response = client.post(
+        "/api/v1/measurements", json=measurements, headers={"X-API-Key": "123"}
+    )
+    assert response.status_code == 403
+
+
+def test_query(client):
+    query = {
+        "start": "1984-04-24T00:00:00",
+    }
+
+    response = client.get(f"/api/v1/measurements?{urlencode(query)}")
+    assert response.status_code == 200
+    assert response.json() == measurements
+
+
+def test_empty_query(client):
+    query = {
+        "source": "test",
+        "start": "1984-04-24T00:00:00",
+        "end": "1984-04-25T00:00:00",
+    }
+
+    response = client.get(f"/api/v1/measurements?{urlencode(query)}")
+    assert response.status_code == 200
+    assert response.json() == []
+
+
+def test_distance_query(client):
+    query = {
+        "start": "1984-04-24T00:00:00",
+        "longitude": -57.521369,
+        "latitude": -25.194156,
+        "distance": "100",
+    }
+
+    response = client.get(f"/api/v1/measurements?{urlencode(query)}")
+    assert response.status_code == 200
+    assert response.json() == measurements
+
+
+def test_enforce_utc():
+    original = measurements[0]
+
+    future = copy.deepcopy(original)
+    future["recorded"] = "2020-10-24T21:47:57.370721+01:00"
+
+    present = copy.deepcopy(original)
+    present["recorded"] = "2020-10-24T20:47:57.370721"
+
+    past = copy.deepcopy(original)
+    past["recorded"] = "2020-10-24T19:47:57.370721-01:00"
+
+    from app.schemas import Measurement
+
+    original = Measurement(**original)
+    future = Measurement(**future)
+    present = Measurement(**present)
+    past = Measurement(**past)
+
+    assert original == future
+    assert original == present
+    assert original == past
+
+
+def test_create_provider(client):
+    response = client.post("/api/v1/providers", json=provider, headers=master_headers)
+
+    assert response.status_code == 200
+
+
+def test_list_providers(client):
+    response = client.get("/api/v1/providers", headers=master_headers)
+
+    assert response.status_code == 200
+    assert response.json() == [provider]
+
+
+def test_delete_provider(client):
+    response = client.delete("/api/v1/providers/test", headers=master_headers)
+
+    assert response.status_code == 200
+
+    response = client.get("/api/v1/providers", headers=master_headers)
+
+    assert response.status_code == 200
+    assert response.json() == []
+
+
+def test_aqi(client):
+    query = {
+        "start": "1984-04-24T00:00:00",
+    }
+
+    response = client.get(f"/api/v1/aqi?{urlencode(query)}")
+    assert response.status_code == 200
+    assert response.json() == aqi
+
+
+def test_stats(client):
+    query = {"start": "1984-04-24T00:00:00"}
+
+    response = client.get(f"/api/v1/stats?{urlencode(query)}")
+    assert response.status_code == 200
+    assert response.json() == stats
+
+
+def test_status(client):
+    response = client.get("/api/v1/status")
+    assert response.status_code == 200
+    assert response.json() == status
```

Comment:
so, yeah, all the tests seem identical to the ones in `services.py`, we should find a way to just do the postgres-based tests on the same `tests/service.py` files.